### DISCUSSION TITLE
twin.macro should support a zero-runtime solution

### DISCUSSION THREAD

**Discussion by NateRadebaugh** [ben-rogerson/twin.macro#244] [Ideas]
To get the most out of my app, I'd like a zero-runtime solution for my CSS. Is this possible with twin.macro? Both `styled-components` and `emotion` have a non-zero runtime size.

**Comment by ben-rogerson** [comment]
A zero runtime is on the roadmap, it's something that I'll be exploring soon enough 👍

**Reply by ben-rogerson** [reply]
what I'd aim for is output to a css/sass file, but it's going to be tricky as there's a css prop/css import involved that twin doesn't have control over.

**Reply by NateRadebaugh** [reply]
Maybe it would be simpler to inject an interop for an existing zero-runtime static css lib like linaria or astroturf?

- https://github.com/callstack/linaria
- https://github.com/4Catalyzer/astroturf

**Reply by ben-rogerson** [reply]
Yeah great ideas here, fits in well with the existing idea of twin being the "glue" between different libs

**Reply by NateRadebaugh** [reply]
I'm not sure how best to test the integrations, I imagine we'll need another example in the [twin.examples](https://github.com/ben-rogerson/twin.examples) repo to validate e2e, but I've got a draft PR here in case this is a direction you want to go in for this:
* https://github.com/ben-rogerson/twin.macro/pull/248
